src/components/TrackTile/TrackerDetails/NumberPicker.tsx

```diff
@@ -0,0 +1,187 @@
+import { Modal, Platform, View, TouchableOpacity } from 'react-native';
+import React, { useCallback, useState } from 'react';
+import { Picker } from '@react-native-picker/picker';
+import { t } from '../../../../lib/i18n';
+import { tID } from '../common/testID';
+import {
+  createStyles,
+  useIcons,
+} from '../../../components/BrandConfigProvider';
+import { Button, Text } from 'react-native-paper';
+import { UnitType } from '../services/TrackTileService';
+import { useStyles } from '../../../hooks';
+
+export type NumberPickerProps = {
+  onChange: (value: string) => void;
+  selectedUnit: UnitType;
+  placeholder?: string;
+  value?: string;
+};
+
+export const NumberPicker = (props: NumberPickerProps) => {
+  const {
+    value = '0',
+    selectedUnit,
+    onChange,
+    placeholder = t('track-tile.set-your-target', 'Set Your Target'),
+  } = props;
+  const { styles } = useStyles(defaultStyles);
+  const [isOpen, setIsOpen] = useState(false);
+  const [tempValue, setTempValue] = useState(value);
+  const { ChevronDown } = useIcons();
+
+  const openInput = useCallback(() => {
+    setIsOpen(true);
+    setTempValue(value);
+  }, [value]);
+
+  const commitNewValue = useCallback(() => {
+    setIsOpen(false);
+    onChange(tempValue);
+  }, [onChange, tempValue]);
+
+  const numbers = [];
+  const stepAmount = selectedUnit.stepAmount ?? 1;
+  const minValue = stepAmount * 1;
+  const maxValue = stepAmount * 50;
```

Comment:
This relies on the stepAmount to be a large enough value that the range of selection is wide enough. For minutes, the step amount is 5 so the selection range is 5-250, which may need to be altered to fit the extreme. This allows a selection of just over 4 hours for trackers like self care and exercise but I could see exercise needing to have a higher upper limit.

Reply:
I'm a little skeptical this will work in all scenarios, and think we should probably take a simpler/dumber approach if possible.  One example that comes to mind is a habit you're trying to break, so we typically want the target to be 0.  The other thing that comes to mind is something like steps where the number is large enough (typical goal of 10k) that the default of 1-50 here will look silly.  Since `stepAmount` and all other props are set on a per-customer, per-metric basis - I'd recommend perhaps attempting these calculations _only_ if more explicit props aren't set (which we'd always recommend setting).  e.g. It could be we need new props (`targetStepAmount`, `targetMin`, `targetMax`), or maybe we have enough props to to reuse?  Either way - I think the preferred way would be 100% config-based, with calculated backups as an edge-case fallback.

Reply:
Thanks, that sounds like a better approach for sure.

Reply:
Done